interp/interp_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"fmt"
+	"math"
+	"testing"
+)
+
+func panics(fn func()) (panicked bool, message string) {
+	defer func() {
+		r := recover()
+		panicked = r != nil
+		message = fmt.Sprint(r)
+	}()
+	fn()
+	return
+}
+
+func TestNewConstInterpolator1D(t *testing.T) {
+	t.Parallel()
+	const value float64 = 42.0
+	i1d := NewConstInterpolator1D(value)
+	if i1d.begin() != math.Inf(-1) {
+		t.Errorf("unexpected begin() value: got: %g want: %g", i1d.begin(), math.Inf(-1))
+	}
+	if i1d.end() != math.Inf(1) {
+		t.Errorf("unexpected end() value: got: %g want: %g", i1d.end(), math.Inf(1))
+	}
+}
+
+func TestConstInterpolator1DEval(t *testing.T) {
+	t.Parallel()
+	const value float64 = 42.0
+	i1d := NewConstInterpolator1D(value)
+	xs := [...]float64{math.Inf(-1), -11, 0.4, 1e9, math.Inf(1)}
```

Comment:
```suggestion
	xs := []float64{math.Inf(-1), -11, 0.4, 1e9, math.Inf(1)}
```
(why create an array? instead of a (more common) slice?)

Reply:
I thought it makes more sense for literal values, but I'll change that.